browser/resources/brave_new_tab_page_refresh/components/settings/top_sites_panel.tsx

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Icon from '@brave/leo/react/icon'
+import Toggle from '@brave/leo/react/toggle'
+
+import { TopSitesListKind } from '../../models/top_sites'
+import { useAppActions, useAppState } from '../context/app_model_context'
+import { useLocale } from '../context/locale_context'
+
+import { style } from './top_sites_panel.style'
+
+export function TopSitesPanel() {
+  const { getString } = useLocale()
+  const actions = useAppActions()
+
+  const showTopSites = useAppState((s) => s.showTopSites)
+  const listKind = useAppState((s) => s.topSitesListKind)
+
+  function listOptionClassName(kind: TopSitesListKind) {
+    if (kind === listKind) {
+      return kind + ' active'
+    }
+    return kind
+  }
+
+  function renderSelectedMarker(kind: TopSitesListKind) {
+    if (kind === listKind) {
+      return (
+        <span className='selected-marker'>
+          <Icon name='check-normal' />
+        </span>
+      )
+    }
+    return null
+  }
+
+  return (
+    <div data-css-scope={style.scope}>
+      <div className='toggle-row'>
+        <label>{getString('showTopSitesLabel')}</label>
+        <Toggle
+          size='small'
+          checked={showTopSites}
+          onChange={({ checked }) => { actions.setShowTopSites(checked) }}
+        />
+      </div>
+      {
+        showTopSites && (
```

Comment:
Hey, this should probably be a `Nala` `ButtonMenu` component.

https://nala.bravesoftware.com/?path=/story/components-buttonmenu--default

if there's a reason it won't work, let's try and update Nala to work in this scenario, rather than adding a new implementation of a common component.

Generally, the Figma files will use the Nala component, and you should just be able to set the same props they're using there.